std/gkr/registry.go

```diff
@@ -0,0 +1,148 @@
+package gkr
+
+import (
+	"fmt"
+	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/std/gkr/internal"
+	"sync"
+)
+
+type GateName string
+
+var (
+	gates     = make(map[GateName]*Gate)
+	gatesLock sync.Mutex
+)
+
+type registerGateSettings struct {
+	solvableVar               int
+	noSolvableVarVerification bool
+	noDegreeVerification      bool
+	degree                    int
+}
+
+// TODO @Tabaie once GKR is moved to gnark, use the same options/settings type for all curves, obviating this
+
+type RegisterGateOption func(*registerGateSettings)
+
+// WithSolvableVar gives the index of a variable whose value can be uniquely determined from that of the other variables along with the gate's output.
+// RegisterGate will return an error if it cannot verify that this claim is correct.
+func WithSolvableVar(solvableVar int) RegisterGateOption {
```

Comment:
Same comment as for gnark-crypto side -- if you don't have the optimizations implemented, then I wouldn't define the options for configuring the solvedness of a variable. I'm not understanding the context.